### PR TITLE
Only include core_types.schema if it's needed

### PR DESCRIPTION
**Pull request by Vatyx**
#### Description
This removes the red warnings we get when launching Spatial since we were including core_types.schema when it didn't need to be.

#### Primary reviewers
@joshuahuburn @improbable-valentyn

### CODE REVIEW

File: SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp [hunk 1]
```diff
@@ -138,16 +138,58 @@ void WriteSchemaRPCField(TSharedPtr<FCodeWriter> Writer, const TSharedPtr<FUnrea
 	);
 }
 
+// core_types.schema should only be included if any components in the file have
+// 1. An UnrealObjectRef
+// 2. A list of UnrealObjectRefs
+// 3. A RPC
```
Comment (improbable-valentyn) [inline SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp]: Nit: `An RPC`

File: SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp [hunk 2]
```diff
@@ -138,16 +138,58 @@ void WriteSchemaRPCField(TSharedPtr<FCodeWriter> Writer, const TSharedPtr<FUnrea
...
+bool ShouldIncludeCoreTypes(TSharedPtr<FUnrealType>& TypeInfo)
+{
+	FUnrealFlatRepData RepData = GetFlatRepData(TypeInfo);
+
+	for (auto& PropertyGroup : RepData)
+	{
+		for (auto& PropertyPair : PropertyGroup.Value)
+		{
+			UProperty* Property = PropertyPair.Value->Property;
+			if(Property->IsA(UObjectPropertyBase::StaticClass()))
```
Comment (improbable-valentyn) [inline SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp]: Nit: space.
Also, personal preference for `Property->IsA<UObjectPropertyBase>()`

File: SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp [hunk 3]
```diff
@@ -138,16 +138,58 @@ void WriteSchemaRPCField(TSharedPtr<FCodeWriter> Writer, const TSharedPtr<FUnrea
...
+			{
+				return true;
+			}
+
+			if (Property->IsA(UArrayProperty::StaticClass()))
```
Comment (improbable-valentyn) [inline SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp]: Nit: Preference for Property->IsA<UObjectPropertyBase>()

File: SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp [hunk 4]
```diff
@@ -138,16 +138,58 @@ void WriteSchemaRPCField(TSharedPtr<FCodeWriter> Writer, const TSharedPtr<FUnrea
...
+			{
+				if (Cast<UArrayProperty>(Property)->Inner->IsA(UObjectPropertyBase::StaticClass()))
```
Comment (improbable-valentyn) [inline SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp]: Nit: preference for `->IsA<UObjectPropertyBase>()`

File: SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp (hunk 1 above)
Comment (Vatyx) [reply SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp]: I'm pretty sure this doesn't follow English rules

**Pull request merged**